rules/Composer/Application/FileProcessor/ComposerFileProcessor.php

```diff
@@ -29,17 +30,14 @@ final class ComposerFileProcessor implements FileProcessorInterface
      */
     private $composerRectors = [];
 
-    /**
-     * @param ComposerRectorInterface[] $composerRectors
-     */
     public function __construct(
         ComposerJsonFactory $composerJsonFactory,
         ComposerJsonPrinter $composerJsonPrinter,
-        array $composerRectors
+        ActiveRectorsProvider $activeRectorsProvider
```

Comment:
I assume this service some kind of filtering based in `SKIP` parameter, right?

If so, is there a use case where we need rectors without this filtering?

It would make sense to put this filtering on [CompilerPass](https://symfony.com/doc/current/service_container/compiler_passes.html) level, so we make sure we alwasy get the same list of rules.

Reply:
Maybe in the future we have some kind of list command for all rules of a given interface. I would not use a CompilerPass here to only get active ones.

Reply:
I'm not sure I understand. I mean now there is a space to forget `ActiveRectorsProvider`, as we use array autowiring convention across the project. We have to keep this convention to avoid [memory locks](https://tomasvotruba.com/blog/2018/08/27/why-and-how-to-avoid-the-memory-lock/).

Here the `array $composerRectors` must remain

Reply:
@TomasVotruba I understand, but as we already use this pattern with the ActiveRectorsProvider for the PhpFileProcessors i thought it is consistent to use it here again. And honestly i am not sure how to do it on CompilerPass level without introducing a lot of complexity here. I haven´t used custom CompilerPasses that much. Could you put me on the right track, maybe i can get somewhere.

Reply:
I implemented the ActiveRectorsProvider very poorly. It should not require so much work to use.

This is 1-place solution with CompilerPass: https://github.com/rectorphp/rector/pull/6286

Reply:
Now you can get all the active Rector rule count like this:

```php
/**
 * @param RectorInterface[]
 */
public function __construct(array $rectors)
{
     $activeRectors = array_filter($rectors, function (RectorInterface $rector) {
         // skip interface post rectors, that are always present
         return ! $rector instanceof PostRectorInterface;
     }

     $activeRectorCount = count($activeRectors);
}
```